lib/matplotlib/tests/test_axes.py

```diff
@@ -2342,6 +2342,23 @@ def test_hist_zorder(histtype, zorder):
         assert patch.get_zorder() == zorder
 
 
+def test_stairs_no_baseline_fill_warns():
+    fig, ax = plt.subplots()
+    baseline = None
+    fill = True
+    warn_match = (
+        "Because baseline in None, the Path used to draw the stairs will "
+    )
+    with pytest.warns(UserWarning, match=warn_match):
+        ax.stairs(
+            [4, 5, 1, 0, 2],
+            [1, 2, 3, 4, 5, 6],
+            facecolor="blue",
+            baseline=baseline,
+            fill=fill
+        )
```

Comment:
Why not inline?
```suggestion
    with pytest.warns(UserWarning, match="baseline=None and fill=True"):
        ax.stairs(
            [4, 5, 1, 0, 2],
            [1, 2, 3, 4, 5, 6],
            facecolor="blue",
            baseline=None,
            fill=True
        )
```

Reply:
a first version of this test was trying to format and test for the whole message including the f-strings but for reasons I don't understand it was not matching so I went with just the middle part of the message with dropped the f-strings.